src/Orleans.Core/Providers/LegacyProviderConfigurator.cs

```diff
@@ -0,0 +1,168 @@
+                    }
+                }
+                else if (providerGroup.Key == ProviderCategoryConfiguration.STORAGE_PROVIDER_CATEGORY_NAME)
+                {
+                    services.AddSingleton<IStorageProvider>(sp => sp.GetServiceByName<IStorageProvider>(ProviderConstants.DEFAULT_STORAGE_PROVIDER_NAME));
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<IStorageProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else if (providerGroup.Key == ProviderCategoryConfiguration.LOG_CONSISTENCY_PROVIDER_CATEGORY_NAME)
+                {
+                    services.AddSingleton<ILogConsistencyProvider>(sp => sp.GetServiceByName<ILogConsistencyProvider>(ProviderConstants.DEFAULT_LOG_CONSISTENCY_PROVIDER_NAME));
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<ILogConsistencyProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else if (providerGroup.Key == ProviderCategoryConfiguration.STATISTICS_PROVIDER_CATEGORY_NAME)
+                {
+                    IProviderConfiguration providerConfig = providerGroup.SelectMany(kvp => kvp.Value.Providers.Values).FirstOrDefault();
+                    if(providerConfig != null)
+                    {
+                        // Looks like we only support a single statistics provider that can be any of the publisher interfaces. fml
+                        // TODO: Kill our statistics system.. please!?
+                        services.AddSingleton<IStatisticsPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as IStatisticsPublisher);
+                        services.AddSingleton<ISiloMetricsDataPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as ISiloMetricsDataPublisher);
+                        services.AddSingleton<IClientMetricsDataPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as IClientMetricsDataPublisher);
+                        RegisterProvider<IProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else
+                {
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<IProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+            }
+        }
+
+        private class ProviderLifecycleParticipant<TService> : ILifecycleParticipant<TLifeCycle>
```

Comment:
This is the wrapper class that allows providers to take part in the silo lifecycle.  It also adds the ability for users to specify the stage they wish the provider to initialize and start in, should they choose.  (not required)